.travis.yml

```diff
@@ -107,11 +107,15 @@ jobs:
       services:
         - docker
       before_install:
-        - docker run -dit --name emscripten -v $(pwd):/src trzeci/emscripten:sdk-incoming-64bit bash
+        # TODO: Emscripten upstream produces a broken build in CI environments only :(
+        - docker run -dit --name emscripten -v $(pwd):/src trzeci/emscripten-fastcomp bash
       script:
         # run binaryen.js tests before and after building, so we see if the bundled
         # version is good too
+        - bash -c 'while sleep 300; do echo ""; done' & PID=$!
         - docker exec -it emscripten bash ./travis-emcc-tests.sh
+        - kill $PID
+        # printing something every 5 minutes prevents Travis from timing out.
```

Comment:
have you seen such timeouts?

Reply:
Yes, compiling Binaryen.js used to time out when I first tried the emscripten-upstream instead of the emscripten-fastcomp image (not here, but in my branch while investigating why upstream builds are broken), because compilation didn't print to console for more than 10 minutes, which Travis considers a stalled build, cancelling it. My expectation is that we will run into this sooner or later, so I thought why not add it.